app/src/main/java/com/example/androidthings/testapp/controleur/Temperature.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.androidthings.testapp.controleur
+
+import android.util.Log
+import com.example.androidthings.testapp.singleton.SerialSingleton
+import org.json.JSONObject
+import org.restlet.data.MediaType
+import org.restlet.ext.json.JsonRepresentation
+import org.restlet.representation.Representation
+import org.restlet.representation.StringRepresentation
+import org.restlet.resource.Get
+import org.restlet.resource.Post
+import org.restlet.resource.ServerResource
+import utils.TAG
+
+/** Controleur to get temperature from android things Device**/
+class Temperature: ServerResource() {
+
+    /** if post :
+     * @param json object
+     * call getTemp function
+     * @return json with temperature
+     **/
+    @Get("")
+    fun getState(entity: JsonRepresentation): Representation {
+        var result = JSONObject()
+        var get = "404 error"
```

Comment:
You shouldn't initialize this variable with your error message and its name seems wrong to me. (Maybe `jsonStatus` is better.)
Otherwise, are you sure that is the way to deal with http 404 error over restlet framework ?